java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1486,7 +1504,7 @@ public Scalar sumOfSquares(DType outType) {
    */
   public Scalar mean() {
     DType outType = DType.FLOAT64;
-    if (type == DType.FLOAT32) {
+    if (type.equals(DType.FLOAT32)) {
```

Comment:
Nit: I thought the code ensured `DType.FLOAT32` and other, non-parameterized types were singletons and therefore legitimate to compare via object equality like an enum.

Reply:
It should, but with java serialization and what not I am not 100% sure in all cases it will always work, so for me, new code should not rely on it, but should use `.equals` if possible.